File: src/main/java/net/miarma/mkernel/util/PlayerUtil.java
package net.miarma.mkernel.util;

import net.miarma.mkernel.MKernel;
import net.miarma.mkernel.common.minecraft.Warp;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.configuration.file.YamlConfiguration;
import org.bukkit.entity.Player;

import java.io.File;
import java.util.Collection;
import java.util.List;

public class PlayerUtil {
    public static double distance(Player p1, Player p2) {
        return p1.getLocation().distance(p2.getLocation());
    }

    public static boolean playersNearRadius(Player player, Collection<Player> players, int radius) {
        return players.stream().anyMatch(p -> distance(p, player) <= radius && !p.equals(player));
    }

    public static List<Warp> getWarps(Player p) {
        File f = new File(MKernel.PLUGIN.getDataFolder().getAbsolutePath(), "warps/"
                + p.getName() + ".yml");
        FileConfiguration c = YamlConfiguration.loadConfiguration(f);
        return c.getKeys(false).stream()
                .map(alias -> Warp.fromFile(c, alias))
                .toList();
    }
}

File: src/main/kotlin/net/miarma/mkernel/util/PlayerUtil.kt
package net.miarma.mkernel.util

import com.destroystokyo.paper.profile.PlayerProfile
import org.bukkit.Bukkit
import org.bukkit.entity.Entity
import org.bukkit.entity.Player

object PlayerUtil {

    fun isWithinRadius(e1: Entity, e2: Entity, radius: Int): Boolean {
        if (e1.world != e2.world) return false
        return e1.location.distanceSquared(e2.location) <= (radius * radius)
    }

    fun playersNearRadius(player: Player, players: Collection<Player>, radius: Int): Boolean {
        return players.any { p -> p != player && isWithinRadius(p, player, radius) }
    }

    fun <E : Entity> isEntityNear(player: Player, entityClass: Class<E>, radius: Int): Boolean {
        return player.location.chunk.entities.any { entity ->
            entityClass.isInstance(entity) && isWithinRadius(player, entity, radius)
        }
    }

    fun fromArg(argumentObject: Any?): Player? {
        return when (argumentObject) {
            is Collection<*> -> {
                val first = argumentObject.firstOrNull()
                (first as? PlayerProfile)?.id?.let { Bukkit.getPlayer(it) }
            }
            else -> null
        }
    }
}
